lib/virt_autotest/utils.pm

```diff
@@ -64,6 +64,23 @@ sub is_pv_guest {
     return $guest =~ /\bpv\b/;
 }
 
+#Check if guest is SLE with optional filter for:
+#Version: <=12-sp3 =12-sp1 >11-sp1 >=15 15+ (>=15 and 15+ are equivalent)
+#usage: guest_is_sle($guest_name, '<=12-sp2')
+sub guest_is_sle {
+    my $guest_name = lc shift;
+    my $query = shift;
+
+    return 0 unless $guest_name =~ /sle/;
+    return 1 unless $query;
+
+    # Version check
+    $guest_name =~ /sles-*(\d{2})(?:-*sp(\d))?/;
+    my $version = $2 eq '' ? "$1-sp0" : "$1-sp$2";
+    return check_version($query, $version, qr/\d{2}(?:-sp\d)?/);
+}
```

Comment:
sle does not mean sles. If you mean sles, then use sles ?

Reply:
This functions looks like check_sles_guest_version to me. And I think the most reliable way to check guest version is get its os_release file and checking from there. I think you can consider using check_os_release in version_utils.pm which supports fetching os_release through ssh.

Reply:
The function name does not reflect what it does. It looks like check-sles-guest-version.

Reply:
I do not think      $guest_name =~ /sles-*(\d{2})(?:-*sp(\d))?/;    is the optimal way to obtain guest version. Consider using check_os_release in version_utils.pm which supports fetching os_releae file and checking version from it. I think os_relase file is the most reliable source to get version info.

Reply:
the widely used lib function, is_sle(), works this way, so I just follow the convention to make it well accepted.

Getting the version from guest name instead of checking inside guest is not so accurate but it has many advantages:

- no need to start guest
- quite easi and quicker